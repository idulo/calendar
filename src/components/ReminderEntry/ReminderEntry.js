import React from "react";
import moment from "moment";

import STYLES from "./ReminderEntry.module.css";

const ReminderEntry = ({ reminder }) => (
  <div className={STYLES.container}>
    On{" "}
    {moment()
      .date(reminder.day)
      .month(reminder.month)
      .format("Do MMMM")}{" "}
    you should: {reminder.text}
  </div>
);

export default ReminderEntry;
