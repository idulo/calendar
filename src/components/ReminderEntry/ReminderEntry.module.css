.container {
  font-size: small;
  margin-bottom: 1rem;
}
