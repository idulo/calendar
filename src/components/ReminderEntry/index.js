import ReminderEntry from "./ReminderEntry";

export default ReminderEntry;
