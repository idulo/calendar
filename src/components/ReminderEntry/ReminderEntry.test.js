import React from "react";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";
import ReminderEntry from "./ReminderEntry";

describe("ReminderList", () => {
  const reminderEntry = { month: 1, day: 1, text: "Merry Christmas" };

  const reminder = it("renders without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(<ReminderEntry reminder={reminderEntry} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly a list of reminders", () => {
    const Picker = create(<ReminderEntry reminder={reminderEntry} />);

    expect(Picker.toJSON()).toMatchSnapshot();
  });
});
