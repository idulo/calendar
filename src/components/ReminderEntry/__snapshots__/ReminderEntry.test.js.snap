// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReminderList renders correctly a list of reminders 1`] = `
<div
  className="container"
>
  On
   
  1st February
   
  you should: 
  Merry Christmas
</div>
`;
